src/layout/mod.rs

```diff
@@ -464,6 +478,56 @@ impl<W: LayoutElement> Layout<W> {
         }
     }
 
+    /// Adds a new window to the layout on a specific output & workspace.
+    pub fn add_window_on_workspace(
+        &mut self,
+        output: &Output,
```

Comment:
Workspace names are globally unique, so this argument should be unnecessary, right?

Reply:
In fact, it *should* be removed, because the output is set in the initial configure, but between the initial configure and the window mapping commit the output could've disappeared, and the window should still open on the named workspace (which is now elsewhere). So this function should return the output where it opened on.

Reply:
Tbh, the single reason I had output here is that then I didn't need to find it again from this function too. I seem to recall having trouble doing that. I don't recall the exact problem, though, so I'll give it another try and report back!